File: C++/Medium/Accepted/rotate-function_2.cpp
class Solution {
public:
    int maxRotateFunction(vector<int>& nums) {
        
        int n = nums.size();
        int sum = 0;
        int F = 0;
        for (int i = 0; i < n; ++i) {
            sum += nums[i];
            F += i * nums[i];
        }
        int maxF = F;
        for (int i = 1; i < n; ++i) {
            F = F + sum - n * nums[n - i];
            maxF = max(maxF, F);
        }
        return maxF;
    }
};

File: Python3/Medium/Accepted/rotate-function_2.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        
        n = len(nums)
        s = sum(nums)
        t = sum(i * v for i, v in enumerate(nums))
        res = t
        for i in range(n - 1, 0, -1):
            t = t + s - n * nums[i]
            res = max(res, t)
        return res